nodePackages.tsun doesn't start up

**Describe the bug**
`nodePackages.tsun` does not start up.

**To Reproduce**
Install and run `tsun` in the terminal.

**Expected behavior**
A TypeScript REPL should start up.

**Additional context**
```
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/nix/store/hpihngi6zpkna60mv4s8mgjk20mfh2kh-node_tsun-0.5.0/lib/node_modules/tsun/bin/src/register.js:3:10)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
```

It's possible that the peer dependency on TypeScript is not handled properly.

**Notify maintainers**
@terlar 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.41, NixOS, 20.09pre226012.b47873026c7 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(brian): `"nixos-20.09pre225673.8ba41a1e149"`
 - channels(root): `"nixos-20.09pre226012.b47873026c7"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

> Ah, it is a bit confusing, but seems you need nodejs as well, this works for me:
> 
> ```
> nix-shell -p nodejs nodePackages.typescript nodePackages.tsun --run tsun
> ```
> 
> It seems without nodejs it won't setup the `NODE_PATH` which is used for the modules to be found.

In `nix-shell` this works for me. But when I have all this in my `configuration.nix` and install a new generation, `tsun` fails with the same error as before.

I've tried having

``` nix
    tsun = super.tsun.override {
      nativeBuildInputs = [ pkgs.makeWrapper ];
      postInstall = ''
      wrapProgram "$out/bin/tsun" \
        --prefix PATH : ${stdenv.lib.makeBinPath [ super.typescript ]}
    '';
    };
```

in `pkgs/development/node-packages/default.nix` but then I get

``` sh
[brian@leungbk-nixos:~]$ tsun
internal/modules/cjs/loader.js:969
  throw err;
  ^

Error: Cannot find module 'typescript'
Require stack:
- /nix/store/v4xsc5kmxgppp0y54p8jlm4m7ri4igcx-node_tsun-0.5.0/lib/node_modules/tsun/bin/src/register.js
- /nix/store/v4xsc5kmxgppp0y54p8jlm4m7ri4igcx-node_tsun-0.5.0/lib/node_modules/tsun/bin/tsun.js
- /nix/store/v4xsc5kmxgppp0y54p8jlm4m7ri4igcx-node_tsun-0.5.0/lib/node_modules/tsun/bin/tsun
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/nix/store/v4xsc5kmxgppp0y54p8jlm4m7ri4igcx-node_tsun-0.5.0/lib/node_modules/tsun/bin    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Module.require (internal/modules/cjs/loader.js:1026:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/nix/store/v4xsc5kmxgppp0y54p8jlm4m7ri4igcx-node_tsun-0.5.0/lib/node_modules/tsun/bin/src/register.js',
    '/nix/store/v4xsc5kmxgppp0y54p8jlm4m7ri4igcx-node_tsun-0.5.0/lib/node_modules/tsun/bin/tsun.js',
    '/nix/store/v4xsc5kmxgppp0y54p8jlm4m7ri4igcx-node_tsun-0.5.0/lib/node_modules/tsun/bin/tsun'
  ]
}
```

after installing from `configuration.nix`.

--- a/pkgs/development/node-packages/default.nix
+++ b/pkgs/development/node-packages/default.nix
@@ -131,6 +131,14 @@ let
       '';
     };
 
+    tsun = super.tsun.overrideAttrs (oldAttrs: {
+      buildInputs = oldAttrs.buildInputs ++ [ pkgs.makeWrapper ];
+      postInstall = ''
+        wrapProgram "$out/bin/tsun" \
+        --prefix NODE_PATH : ${self.typescript}/lib/node_modules
+      '';
+    });
+
     stf = super.stf.override {
       meta.broken = since "10";
     };

